JitsiConference.ts

```diff
@@ -748,3153 +755,3253 @@ export default class JitsiConference extends Listenable {
-                reason,
-                reasonDescription,
-                sendSessionTerminate: this.room
-                    && this.getParticipantById(
-                        Strophe.getResourceFromJid(this.p2pJingleSession.remoteJid))
-            });
 
-        this.p2pJingleSession = null;
+                try {
+                    messageToSend = JSON.stringify(messageToSend);
+                } catch (e) {
+                    logger.error('Can not send a message, stringify failed: ', e);
 
-        // Update P2P status and other affected events/states
-        this._setP2PStatus(false);
+                    return;
+                }
+            }
 
-        if (wasP2PEstablished) {
-            // Add back remote JVB tracks
-            if (this.jvbJingleSession && !requestRestart) {
-                this._addRemoteJVBTracks();
+            if (to) {
+                this.sendPrivateTextMessage(to, messageToSend, elementName);
             } else {
-                logger.info('Not adding remote JVB tracks - no session yet');
+                // Broadcast
+                this.sendTextMessage(messageToSend, elementName);
             }
         }
     }
 
     /**
-     * Updates room presence if needed and send the packet in case of a modification.
-     * @param {JingleSessionPC} jingleSession the session firing the event, contains the peer connection which
-     * tracks we will check.
-     * @param {Object|null} ctx a context object we can distinguish multiple calls of the same pass of updating tracks.
+     * Checks if the connection is interrupted.
+     * @returns {boolean} True if the connection is interrupted, false otherwise.
      */
-    _updateRoomPresence(jingleSession: JingleSessionPC, ctx: { skip?: boolean; } = {}): void {
-        if (!jingleSession) {
-            return;
-        }
-
-        // skips sending presence twice for the same pass of updating ssrcs
-        if (ctx) {
-            if (ctx.skip) {
-                return;
-            }
-            ctx.skip = true;
-        }
-
-        let presenceChanged = false;
-        let muteStatusChanged: boolean;
-        let videoTypeChanged: boolean;
-        const localTracks = jingleSession.peerconnection.getLocalTracks();
-
-        // Set presence for all the available local tracks.
-        for (const track of localTracks) {
-            const muted = track.isMuted();
-
-            muteStatusChanged = this._setTrackMuteStatus(track, muted);
-            muteStatusChanged && logger.debug(`Updating mute state of ${track} in presence to muted=${muted}`);
-            if (track.getType() === MediaType.VIDEO) {
-                videoTypeChanged = this._setNewVideoType(track);
-                videoTypeChanged && logger.debug(`Updating videoType in presence to ${track.getVideoType()}`);
-            }
-            presenceChanged = presenceChanged || muteStatusChanged || videoTypeChanged;
-        }
+    public isConnectionInterrupted(): boolean {
```

Comment:
I think we can drop this function, it's actually unused. Jitsi Meet calls it but nothing is calling the function JM defines.